### PR TITLE
add xgboost mnist example for pyspark

### PR DESCRIPTION
**Pull request by andremoeller**
This is effectively identical to the KMeans example here, but uses XGBoost instead. https://github.com/awslabs/amazon-sagemaker-examples/pull/168

### CODE REVIEW

File: sagemaker-spark/pyspark_mnist/xgboost_mnist.ipynb [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from pyspark import SparkContext, SparkConf\n",
+    "from pyspark.sql import SparkSession\n",
+    "import os\n",
+    "import sagemaker_pyspark\n",
+    "import sagemaker\n",
+    "from sagemaker import get_execution_role\n",
```
Comment (laurenyu) [inline sagemaker-spark/pyspark_mnist/xgboost_mnist.ipynb]: nitpick - same thing as before about import statements: Python standard library, then third-party libraries, then local ones

File: sagemaker-spark/pyspark_mnist/xgboost_mnist.ipynb [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    "    plt.xlabel(xlabel)\n",
+    "    subplot.imshow(imgr, cmap='gray')\n",
+    "\n",
+    "images = np.array(transformedData.select(\"features\").cache().take(250))\n",
+    "clusters = transformedData.select(\"prediction\").cache().take(250)\n",
+    "\n",
+    "for cluster in range(10):\n",
+    "    print('\\n\\n\\nCluster {}:'.format(int(cluster)))\n",
+    "    digits = [ img for l, img in zip(clusters, images) if int(l.prediction) == cluster ]\n",
+    "    height=((len(digits)-1)//5)+1\n",
```
Comment (laurenyu) [inline sagemaker-spark/pyspark_mnist/xgboost_mnist.ipynb]: there's a [pep8 recommendation](https://www.python.org/dev/peps/pep-0008/#other-recommendations) to put whitespace around operators with the lowest priority, i.e.
```python
((len(digits) - 1)//5) + 1
```
YMMV.

File: sagemaker-spark/pyspark_mnist/xgboost_mnist.ipynb [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Delete the endpoint\n",
```
Comment (laurenyu) [inline sagemaker-spark/pyspark_mnist/xgboost_mnist.ipynb]: same suggestion as before - adding a short text cell to explain why we're deleting the endpoint would be nice

**Comment by andremoeller** [comment]
Done, thanks Lauren!

**Comment by andremoeller** [comment]
Awesome, thanks David! I've made the changes, same as https://github.com/awslabs/amazon-sagemaker-examples/pull/168

**Pull request merged**